pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -384,7 +384,7 @@ stdenv.mkDerivation ({
     done
 
     for d in $(grep '^dynamic-library-dirs:' "$packageConfDir"/* | cut -d' ' -f2- | tr ' ' '\n' | sort -u); do
-      for lib in "$d/"*.dylib; do
+      for lib in "$d/"*.{dylib,so}; do
```

Comment:
${stdenv.hostPlatform.extensions.sharedLibrary}

Reply:
Thanks for the review @FRidh! Just to clarify, you're suggesting it should be this:
```
- for lib in "$d/"*.dylib; do
+ for lib in "$d/"*{${stdenv.hostPlatform.extensions.sharedLibrary},.so}; do
```
right? Or maybe a bit clearer like this:
```
- for lib in "$d/"*.dylib; do
+ for lib in "$d/"*${stdenv.hostPlatform.extensions.sharedLibrary} "$d/"*.so; do
```
I'm happy to do either of these, but to my eyes this makes the code more obfuscated. This whole section of code is intensely Darwin-specific, and the code has to know about both `.so` and `.dylib`, so there's no generality gained.

So can I just check that this is what you think would be an improvement? If so I'll happily re-test and push.

Reply:
stdenv.hostPlatform.extensions.sharedLibrary evaluates to .so on Linux and .dylib on Darwin. I doubt you need to perform an action on both .dylib and .so *in the same build*.

Reply:
Unfortunately, that is exactly the point of this patch. See #80190 for more details.

I share your intuition that there should be a better way, and would love someone to propose a superior alternative. Right now though, this is the only fix I have for the broken packages.

Reply:
BTW the Darwin nix-store in front of me (no cross-compilation) has 44421 `.dylib`s and 778 `.so`s. Of those `.so`s, there are two distinct ones that `otool -hv` reports as `DYLIB` rather than `BUNDLE`:

 - the `libstdbuf.so` from coreutils
 - the `liblmdb.so` from lmdb at issue here

I have no idea whether the real root cause for this issue is something to do with them, but as it stands, there definitely are dylibs with suffix `.so` on Darwin.

Reply:
Turns out the liblmdb `.so` suffix was specified [manually](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/lmdb/default.nix#L26) in 84bd2f4ab04b8b53718db6b5ba208a62ae083941 over two years ago.
I still do not understand why 8d4509e34d85501dca2eed1a2dfbd5c0611b0f5b from a couple of weeks ago should trigger a breakage though.

Reply:
The install_name needs to point to the full path to the final library location. Perhaps, the library got moved as a result of 8d4509e34d85501dca2eed1a2dfbd5c0611b0f5b so the line
https://github.com/NixOS/nixpkgs/blob/82945f364faa580a7c701c83d9c6c5cc9b7ef2fe/pkgs/development/libraries/lmdb/default.nix#L26
needs to be updated to reflect the move.

Reply:
Good thought @veprbl - I've looked, and to the best of my understanding, this isn't the diagnosis. Both working and broken builds end up depending on the exact same LMDB C package (same SHA) where `otool -L` does indeed show the full path to the final library location.